### PR TITLE
add ES function

### PR DESCRIPTION
**Pull request by cooper-lzy**

### CODE REVIEW

File: docs-2.0/3.ngql-guide/6.functions-and-expressions/17.ES-function.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+```ngql
+calles("<es_query>");
+```
+
+- 返回类型：map。
+
+## 示例
+
+```ngql
+nebula> WITH calles("test/player/1") AS a RETURN a;
```
Comment (nevermore3) [inline docs-2.0/3.ngql-guide/6.functions-and-expressions/17.ES-function.md:30]: arg's format 's  /<indexName>/<typeName>/<id>

File: docs-2.0/3.ngql-guide/6.functions-and-expressions/17.ES-function.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
++---------------------------------------------------------------------------------------------------------------------------------------------+
+| a                                                                                                                                           |
++---------------------------------------------------------------------------------------------------------------------------------------------+
+| {_id: "1", _index: "test", _primary_term: 1, _seq_no: 0, _source: {age: 43, name: "Tim Duncan"}, _type: "player", _version: 1, found: true} |
++---------------------------------------------------------------------------------------------------------------------------------------------+
+
+nebula> WITH calles("test/player/1") AS a MATCH (b:player) WHERE b.player.age == a["_source"]["age"] RETURN b;
```
Comment (nevermore3) [inline docs-2.0/3.ngql-guide/6.functions-and-expressions/17.ES-function.md:37]: dito

File: docs-2.0/3.ngql-guide/6.functions-and-expressions/17.ES-function.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
++---------------------------------------------------+
+| b                                                 |
++---------------------------------------------------+
+| ("player141" :player{age: 43, name: "Ray Allen"}) |
++---------------------------------------------------+
```
Comment (nevermore3) [inline docs-2.0/3.ngql-guide/6.functions-and-expressions/17.ES-function.md:42]: pls add `callESFile('<filePath>')` to the case

File: docs-2.0/3.ngql-guide/6.functions-and-expressions/17.ES-function.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# Elasticsearch 查询函数
+
+calles() 函数可以通过 GET 请求执行复杂的查询语句查询 ES 的数据。
+
+关于 ES 语法，请参见 [ES 官方文档](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-your-data.html)。
```
Comment (whitewum) [inline docs-2.0/3.ngql-guide/6.functions-and-expressions/17.ES-function.md:5]: ES不要简写

**Review by whitewum — CHANGES_REQUESTED** [review]
how to prepare data?

File: docs-2.0/3.ngql-guide/6.functions-and-expressions/17.ES-function.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Elasticsearch 查询函数
+
+calles() 函数可以通过 GET 请求执行复杂的查询语句查询 ES 的数据。
```
Comment (whitewum) [inline docs-2.0/3.ngql-guide/6.functions-and-expressions/17.ES-function.md:3]: 完全看不懂，用个逗号分割两句话吧。。。。

**Review by whitewum — CHANGES_REQUESTED** [review]

**Review by whitewum — APPROVED** [review]

**Review by whitewum — APPROVED** [review]

File: docs-2.0/3.ngql-guide/6.functions-and-expressions/17.ES-function.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+- 使用前需要 SIGN IN 服务，例如`SIGN IN TEXT SERVICE (127.0.0.1:9200, HTTP)`，不再使用时可以 SIGN OUT 服务。具体语法请参见[登录文本搜索客户端](../../4.deployment-and-installation/6.deploy-text-based-index/2.deploy-es.md#_3)。
+
+- 只支持 GET 请求。请用 Elasticsearch 客户端执行 POST、PUT、DELETE 请求。
+
+- 没有超时时间，Nebula Graph 会一直等待 Elasticsearch 返回结果。
+
+- 不支持函数后直接跟操作符，例如`RETURN calles("/test/player/1")["_source"]`。
+
+- 兼容 Elasticsearch 7及以上版本，更低版本未测试。
```
Comment (abby-cyber) [inline docs-2.0/3.ngql-guide/6.functions-and-expressions/17.ES-function.md:23]:
```suggestion
- 兼容 Elasticsearch 7 及以上版本，更低版本未测试。
```

File: docs-2.0/4.deployment-and-installation/6.deploy-text-based-index/1.text-based-index-restrictions.md [hunk 7]
```diff
@@ -22,6 +22,8 @@
 
 - 全文索引不支持搜索属性值为`NULL`的属性。
 
+- 写入数据时，Elastic Search 会自动创建相应的索引。在 Nebula Graph 中创建全文索引，但是未写入数据，查询时会报错`text search not found`。
```
Comment (abby-cyber) [inline docs-2.0/4.deployment-and-installation/6.deploy-text-based-index/1.text-based-index-restrictions.md:25]:
```suggestion
- 写入数据时，Elasticsearch 会自动创建相应的索引。在 Nebula Graph 中创建全文索引，但是未写入数据，查询时会报错`text search not found`。
```

Comment (abby-cyber) [reply docs-2.0/4.deployment-and-installation/6.deploy-text-based-index/1.text-based-index-restrictions.md:25]: Elastic Search--->Elasticsearch?

File: docs-2.0/4.deployment-and-installation/6.deploy-text-based-index/1.text-based-index-restrictions.md [hunk 8]
```diff
@@ -22,6 +22,8 @@
...
+
 - 不支持修改 Elastic Search 中的索引，只能删除重建。
```
Comment (abby-cyber) [inline docs-2.0/4.deployment-and-installation/6.deploy-text-based-index/1.text-based-index-restrictions.md:27]:
```suggestion
- 不支持修改 Elasticsearch 中的索引，只能删除重建。
```

File: mkdocs.yml [hunk 9]
```diff
@@ -298,6 +298,7 @@ nav:
           - 条件表达式函数: 3.ngql-guide/6.functions-and-expressions/5.conditional-expressions.md
           - 谓词函数: 3.ngql-guide/6.functions-and-expressions/8.predicate.md
           - geo 函数: 3.ngql-guide/6.functions-and-expressions/14.geo.md
+#          - Elasticsearch 查询函数: 3.ngql-guide/6.functions-and-expressions/17.ES-function.md
```
Comment (abby-cyber) [inline mkdocs.yml:301]: why do we hide this doc?

**Review by abby-cyber — APPROVED** [review]

**Review by foesa-yang — APPROVED** [review]

**Pull request merged**